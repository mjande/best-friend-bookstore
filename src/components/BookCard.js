import { useState } from "react";
import "../styles/BookCard.css";

const BookCard = ({ book, addToCart, removeFromCart }) => {
  const [inCart, toggleInCart] = useState(false);

  const buttonClick = (e) => {
    toggleInCart(!inCart);

    if (inCart) {
      removeFromCart(e);
    } else {
      addToCart(e);
    }
  };

  return (
    <article className="BookCard">
      <img
        src={`https://covers.openlibrary.org/b/olid/${book.cover_edition_key}.jpg`}
        alt="Book cover"
      ></img>
      <div className="BookCard-details">
        <h3>{book.title}</h3>
        <h4>by {book.authors[0].name}</h4>
      </div>
      <button
        onClick={buttonClick}
        data-work-id={book.key}
        data-edition-id={book.cover_edition_key}
        className={inCart ? "in-cart" : "not-in-cart"}
      >
        {inCart ? "Remove from Cart" : "Add to Cart"}
      </button>
    </article>
  );
};

export default BookCard;
