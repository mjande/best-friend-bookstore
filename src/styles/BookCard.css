.BookCard {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 0.5rem;
  width: 200px;
}

.BookCard img {
  border: solid 0.5px;
  height: 200px;
  width: 150px;
  object-fit: cover;
}

.BookCard-details {
  text-align: center;
}

.BookCard button.not-in-cart {
  background-color: #4e9af1;
}

.BookCard button.in-cart {
  background-color: #f14e4e;
}
