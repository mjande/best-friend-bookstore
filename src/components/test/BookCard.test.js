import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import BookCard from "../BookCard";

describe("BookCard", () => {
  it("renders book with image, title, author", () => {
    const { container } = render(
      <BookCard title="Test Title" author="Test Author" image="#" />
    );

    const imageElement = screen.getByRole("img", { name: "Book cover" });
    const titleElement = screen.getByRole("heading", { name: "Test Title" });
    const authorElement = screen.getByRole("heading", {
      name: "by Test Author",
    });

    expect(imageElement).toBeInTheDocument();
    expect(titleElement).toBeInTheDocument();
    expect(authorElement).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
