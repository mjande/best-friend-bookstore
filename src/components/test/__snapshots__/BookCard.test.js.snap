// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard renders book with image, title, author 1`] = `
<div>
  <article
    class="Book"
  >
    <img
      alt="Book cover"
      src="https://covers.openlibrary.org/b/id/undefined.jpg"
    />
    <div
      class="Book-details"
    >
      <h3>
        Test Title
      </h3>
      <h4>
        by 
        Test Author
      </h4>
    </div>
  </article>
</div>
`;
